index.js

```diff
@@ -1,29 +1,92 @@
-module.exports = (opts = { }) => {
+const { writeFile } = require("fs");
 
-  // Work with options here
-  console.log(opts)
+const { getCustomProp, noop } = require("./helper");
 
-  return {
-    postcssPlugin: 'postcss-customprop-validate',
-    /*
-    Root (root, postcss) {
-      // Transform CSS AST here
-    }
-    */
-
-    /*
-    Declaration (decl, postcss) {
-      // The faster way to find Declaration node
-    }
-    */
-
-    /*
-    Declaration: {
-      color: (decl, postcss) {
-        // The fastest way find Declaration node if you know property name
-      }
-    }
-    */
+module.exports = validateCustomProps;
+module.exports.postcss = true;
+
+/**
+ * Object to define an incorrect CSS custom property
+ *
+ * @typedef {Object} WrongProp
+ * @property {string} path - absolute path of the CSS file
+ * @property {number} line - line number containing the custom property
+ * @property {string} key - key of the custom property
+ * @property {string} current - current value of the custom property
+ * @property {string} expexted - expected correct value of the custom property
+ */
+
+/**
+ * Callback to handle the response of the plugin
+ *
+ * @callback requestCallback
+ * @param {Object} error - an error object
+ * @param {WrongProp[]} data - a list of incorrect custom properties
+ * @returns
+ */
+
+/**
+ * Postcss plugin to validate fallback value of CSS custom properties
+ *
+ * @param {Object} opts - plugin options
+ * @param {Object} opts.properties - CSS custom properties to validate values from
+ * @param {boolean} opts.write [write=false] - update the file with updated CSS
+ * @param {requestCallback} opts.callback [callback=() => {}] - callback to handle the response
+ * @returns Object
+ */
+function validateCustomProps({ properties, write = false, callback = noop }) {
+  if (!properties) {
+    throw new Error("No properties present in options");
   }
+
+  return {
+    postcssPlugin: "postcss-customprop-validate",
+    prepare(result) {
+      const wrongProps = [];
+
+      return {
+        Declaration(decl) {
+          const customProp = getCustomProp(decl);
+
+          // exit if the declaration is not a custom property
```

Comment:
@abhinavpreetu how about this comment instead?

`only process custom properties`

the comment should talk about the larger why. not what the code is doing.

Reply:
understood